Fastly/fastly-waf/ingest/parser.yml

```
@@ -0,0 +1,50 @@
+name: fastly-waf
+pipeline:
+  - name: parse_event
+    external:
+      name: json.parse-json
+      properties:
+        input_field: "{{original.message}}"
+        output_field: message
+
+  - name: set_fields
+
+stages:
+  set_fields:
+    actions:
+      - set:
+          "@timestamp": "{{parse_event.message.timestamp}}"
+
+          observer.vendor: "Fastly"
+          observer.product: "Fastly Next-Gen WAF"
+
+          event.kind: "event"
+          event.module: "fastly.waf"
+          event.category: "network"
+
+      - translate:
+        dictionary:
+          "flagged": ["denied"]
+          "info": ["info"]
+        fallback: ["info"]
+        mapping:
+          parse_event.message.action: event.type
+
+      - set:
+          user_agent.original: "{{parse_event.message.userAgents[0]}}"
+        filter: "{{parse_event.message.userAgents != null }}"
+
+      - set:
+          action.name: "{{parse_event.message.action}}"
```

`event.action` seems to be more suitable (standard field).
```suggestion
          event.action: "{{parse_event.message.action}}"
```